app/controllers/coupons_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CouponsController < ApplicationController
+  def index
+    @coupons = Coupon.all
+  end
+
+  def new
+    @coupon = Coupon.new
+  end
```

Comment:
Since `@coupon` isn't being used in the view, it can be omitted here:
```suggestion
  def new; end
```